tests/integration/galley/validation_test.go

```diff
@@ -86,25 +86,23 @@ func TestValidation(t *testing.T) {
 			}
 
 			for _, d := range dataset {
-				t.Run(string(d), func(t *testing.T) {
+				ctx.NewSubTest(string(d)).Run(func(ctx framework.TestContext) {
 					if d.isSkipped() {
 						t.SkipNow()
 						return
 					}
 
-					fctx := framework.NewContext(t)
-					defer fctx.Done()
-
 					ym, err := d.load()
 					if err != nil {
 						t.Fatalf("Unable to load test data: %v", err)
 					}
 
-					ns := namespace.NewOrFail(t, fctx, namespace.Config{
+					ns := namespace.NewOrFail(t, ctx, namespace.Config{
 						Prefix: "validation",
 					})
 
-					_, err = cluster.ApplyContentsDryRun(ns.Name(), ym)
+					applyFiles := ctx.WriteYAMLOrFail(ctx, ym, "apply")
+					err = cluster.ApplyYAMLFiles(ns.Name(), true, applyFiles...)
```

Comment:
why write to a file? applying a string directly is a better ux to me. Does the cluster API still allow this - looks like yes?

Reply:
This one does a dry run, which I believe is the only place in the code that uses dry run today. The ConfigManager API doesn't support dry run, so writing to files first is necessary here. I could work dry run into the ConfigManager API but I'm not sure it's worth it.